### PR TITLE
Adding Image entity page

### PR DESCRIPTION
**Pull request by muthurajendran**
@ThomasSchellenbergNextCentury I am not sure of how to get the blur image config in the main enity html for the image. For now its blur always. Let me know if there is a way we can get them.

### CODE REVIEW

File: app/image.html [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+            <!-- Left Pane -->
+            <div class="flex layout vertical">
+
+              <!-- Timeline - Offers between date1 and date2 with Zoomable Barchart-->
+
+
+              <paper-material>
+                <section class="when">
+                  <iron-image style="width:400px; height:400px; background-color: lightgray;"
+                    sizing="cover" preload fade src="{{imageDisplay.url}}" blurImages class="small-blur"></iron-image>
```
Comment (ThomasSchellenbergNextCentury) [inline app/image.html]: The iron-image should have its sizing property set to "contain" else images that aren't completely square will get cut-off (for example see the image http://dig.isi.edu/ht/data/0F7DC7B22098A8DBF1E259EB712EADF67A71EB34 in the dig-latest index on 52.36.12.77:8080).

Also, if you want the image to be more blurry, change the class from "small-blur" to "large-blur".

File: app/scripts/image-transform.js [hunk 2]
```diff
@@ -24,9 +24,45 @@ var imageTransform = (function(_) {
...
+    image: function(data) {
+      var newData = {};
+      if(data && data.hits.hits.length > 0) {
+        newData = data.hits.hits[0]._source;
+
+        var adultService = {
+          total: newData.isImagePartOf.length,
+          array: []
+        };
+        newData.isImagePartOf.forEach(function(service) {
```
Comment (ThomasSchellenbergNextCentury) [inline app/scripts/image-transform.js:37]: I found an image where isImagePartOf was an object, not an array.  You should check if isImagePartOf is an array and, if not, convert it to one like this:
newData.isImagePartOf = _.isArray(newData.isImagePartOf) ? newData.isImagePartOf : [newData.isImagePartOf];

File: app/image.html [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+     data-in="[[imageResults]]"
+     data-out="{{imageDisplay}}"
+     transform-function="[[clientConfig.transforms.image.image]]"
+     ></json-transform>
+
+     <!-- Get the seller Object -->
+     <query-builder
+      query="{{sellerQuery}}"
+      query-template="{{config.queryTemplates.commonMatchQuery}}"
+      field="[[params.field]]" value="[[phoneDisplay.telephone.sellerId]]">
```
Comment (ThomasSchellenbergNextCentury) [inline app/image.html]: There are references to phoneDisplay.telephone on lines 90, 116, and 220, but I don't think it's getting set anywhere.  Should those be references to imageDisplay instead?

File: app/image.html [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+    <div class="entity">
+
+      <paper-header-panel class="flex" mode="waterfall-tall">
+
+        <!-- Entity header -->
+        <section class="paper-header layout horizontal">
+
+          <!-- Entity title -->
+          <div class="layout horizontal center flex">
+            <iron-icon class="iron-icon-huge entity-phone-font" icon="image:image"></iron-icon>
```
Comment (ThomasSchellenbergNextCentury) [inline app/image.html]: Should be "entity-photo-font" instead of "entity-phone-font".

File: app/image.html [hunk 5]
```diff
@@ -0,0 +1,452 @@
...
+              </div>
+            </section>
+          </div>
+          <annotate-record client="[[esclient]]"
+            annotation-index="[[config.annotationIndex]]"
+            annotation-type="[[config.annotationType]]"
+            annotation-relevant="[[config.annotationRelevant]]"
+            query-template="{{config.queryTemplates.annotationQuery}}"
+            item="{{phoneDisplay.telephone}}"
+            item-type="telephone number"
```
Comment (ThomasSchellenbergNextCentury) [inline app/image.html:193]: Should be "image" instead of "telephone number".

File: server/config/query-templates.js [hunk 6]
```diff
@@ -89,6 +95,65 @@ module.exports = {
         }
       }
     },
+    phoneOrEmailPeopleAggForImages: {
```
Comment (ThomasSchellenbergNextCentury) [inline server/config/query-templates.js]: I don't think phoneOrEmailPeopleAggForImages or sellerPeopleAggs are being used.  If so, you can delete them.

File: app/scripts/image-transform.js [hunk 7]
```diff
@@ -24,9 +24,45 @@ var imageTransform = (function(_) {
       }
       return images;
     },
+    image: function(data) {
```
Comment (ThomasSchellenbergNextCentury) [inline app/scripts/image-transform.js:27]: This function should be tested in client-config-image-test to ensure that the returned object (with all of its properties) is created correctly.

**Comment by muthurajendran** [comment]
@ThomasSchellenbergNextCentury I've fixed all the issues!

**Comment by ThomasSchellenbergNextCentury** [comment]
Great, thanks!

**Pull request merged**